analyzers/src/SonarAnalyzer.CSharp/Rules/MethodsShouldNotHaveIdenticalImplementations.cs

```diff
@@ -29,51 +28,35 @@
-        protected override GeneratedCodeRecognizer GeneratedCodeRecognizer => CSharpGeneratedCodeRecognizer.Instance;
-
-        protected override SyntaxKind ClassDeclarationSyntaxKind => SyntaxKind.ClassDeclaration;
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        protected override IEnumerable<IMethodDeclaration> GetMethodDeclarations(SyntaxNode node)
-            => ((ClassDeclarationSyntax)node).GetMethodDeclarations();
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override bool AreDuplicates(IMethodDeclaration firstMethod, IMethodDeclaration secondMethod)
+        protected override SyntaxKind[] SyntaxKinds { get; } =
         {
-            return firstMethod.Body != null &&
-                secondMethod.Body != null &&
-                firstMethod.Body.Statements.Count >= 2 &&
-                firstMethod.Identifier.ValueText != secondMethod.Identifier.ValueText &&
-                HaveSameParameters(firstMethod.ParameterList?.Parameters, secondMethod.ParameterList?.Parameters) &&
-                firstMethod.Body.IsEquivalentTo(secondMethod.Body, false);
+            SyntaxKind.ClassDeclaration,
+            SyntaxKindEx.RecordDeclaration,
+        };
 
-            static bool HaveSameParameters(SeparatedSyntaxList<ParameterSyntax>? leftParameters, SeparatedSyntaxList<ParameterSyntax>? rightParameters)
-            {
-                if (leftParameters == null && rightParameters == null)
-                {
-                    return true;
-                }
+        protected override GeneratedCodeRecognizer GeneratedCodeRecognizer => CSharpGeneratedCodeRecognizer.Instance;
 
-                if (leftParameters == null ||
-                    rightParameters == null ||
-                    leftParameters.Value.Count != rightParameters.Value.Count)
-                {
-                    return false;
-                }
+        protected override IEnumerable<IMethodDeclaration> GetMethodDeclarations(SyntaxNode node) =>
+            ((TypeDeclarationSyntax)node).GetMethodDeclarations();
 
-                return leftParameters.Value
-                    .Zip(rightParameters.Value, (p1, p2) => new { p1, p2 })
-                    .All(tuple => tuple.p1.IsEquivalentTo(tuple.p2, false));
-            }
-        }
+        protected override bool AreDuplicates(IMethodDeclaration firstMethod, IMethodDeclaration secondMethod) =>
+            firstMethod.Body != null
+            && firstMethod.Body.Statements.Count > 1
+            && firstMethod.Identifier.ValueText != secondMethod.Identifier.ValueText
+            && HaveSameParameters<ParameterSyntax>(firstMethod.ParameterList.Parameters, secondMethod.ParameterList.Parameters)
+            && firstMethod.Body.IsEquivalentTo(secondMethod.Body, false);
```

Comment:
The second parameter is redundant as the default value of topLevel is false.

Reply:
Nope, it's FP. I've already checked that, since issue was raised. If you will remove it, it will call completely different method